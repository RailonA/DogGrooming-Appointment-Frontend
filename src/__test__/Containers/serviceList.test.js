import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import ServiceList from '../../Containers/serviceList';

describe('serviceList', () => {
  const services = [
    {
      id: 1,
      category: 'dog',
      petService: 'Bath & Full Haircut',
      serviceDescription: 'A haircut and style. Service includes Bath services plus a full haircut or shave.',
      servicePrice: '76.00',

    },
    {
      id: 2,
      category: 'dog',
      petService: 'Bath & Full Haircut with FUR minator',
      serviceDescription: 'Helps reduce shedding. Package includes Bath and Haircut services plus Low-Shed Shampoo, DeShedding Solution and Up to 20 Minutes Brushing with FURminator tool.',
      servicePrice: '91.00',

    },
    {
      id: 3,
      category: 'dog',
      petService: 'Bath & Brush',
      serviceDescription: 'Helps reduce shedding. For dogs that do not need a haircut. Package includes Bath services plus Low-Shed Shampoo, DeShedding Solution and Up to 20 Minutes Brushing with FURminator tool.',
      servicePrice: '81.00',
    },
  ];
  const category = 'dog';

  it('should match the snapshot', () => {
    const wrapper = renderer.create(
      <ServiceList services={services} category={category} />,
    ).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
