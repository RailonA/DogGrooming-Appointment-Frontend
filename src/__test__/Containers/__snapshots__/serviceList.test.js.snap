// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serviceList should match the snapshot 1`] = `
<div
  className="container-fluid flex-column justify-content-center"
>
  <div
    className="card"
    onClick={[Function]}
  >
    <table
      className="  m-3 col-10"
    >
      <tbody
        className=" col-12"
      >
        <tr
          className=" m-3 "
        >
          <div
            className="m-3 col-12 flex-container justify-content-center"
          >
            <div
              className="col-12 d-flex justify-content-center"
            >
              <h4
                className=" col-8  d-flex justify-content-center serviceCategoryTitle p-2"
              >
                Bath & Full Haircut
              </h4>
            </div>
            <hr
              className=" col-10 "
            />
          </div>
          <div
            className="d-flex m-4 col-10 justify-content-center serviceCategoryBody"
          >
            <td
              className=" col-9  text-center"
            >
              A haircut and style. Service includes Bath services plus a full haircut or shave.
            </td>
            <td
              className="d-flex col-1 text-center justify-content-center align-self-center serviceCategoryPrice"
            >
              <p
                className="mr-2"
              >
                $
              </p>
              <p>
                76.00
              </p>
            </td>
          </div>
        </tr>
      </tbody>
      <hr
        className=" col-12 "
      />
    </table>
  </div>
  <div
    className="card"
    onClick={[Function]}
  >
    <table
      className="  m-3 col-10"
    >
      <tbody
        className=" col-12"
      >
        <tr
          className=" m-3 "
        >
          <div
            className="m-3 col-12 flex-container justify-content-center"
          >
            <div
              className="col-12 d-flex justify-content-center"
            >
              <h4
                className=" col-8  d-flex justify-content-center serviceCategoryTitle p-2"
              >
                Bath & Full Haircut with FUR minator
              </h4>
            </div>
            <hr
              className=" col-10 "
            />
          </div>
          <div
            className="d-flex m-4 col-10 justify-content-center serviceCategoryBody"
          >
            <td
              className=" col-9  text-center"
            >
              Helps reduce shedding. Package includes Bath and Haircut services plus Low-Shed Shampoo, DeShedding Solution and Up to 20 Minutes Brushing with FURminator tool.
            </td>
            <td
              className="d-flex col-1 text-center justify-content-center align-self-center serviceCategoryPrice"
            >
              <p
                className="mr-2"
              >
                $
              </p>
              <p>
                91.00
              </p>
            </td>
          </div>
        </tr>
      </tbody>
      <hr
        className=" col-12 "
      />
    </table>
  </div>
  <div
    className="card"
    onClick={[Function]}
  >
    <table
      className="  m-3 col-10"
    >
      <tbody
        className=" col-12"
      >
        <tr
          className=" m-3 "
        >
          <div
            className="m-3 col-12 flex-container justify-content-center"
          >
            <div
              className="col-12 d-flex justify-content-center"
            >
              <h4
                className=" col-8  d-flex justify-content-center serviceCategoryTitle p-2"
              >
                Bath & Brush
              </h4>
            </div>
            <hr
              className=" col-10 "
            />
          </div>
          <div
            className="d-flex m-4 col-10 justify-content-center serviceCategoryBody"
          >
            <td
              className=" col-9  text-center"
            >
              Helps reduce shedding. For dogs that do not need a haircut. Package includes Bath services plus Low-Shed Shampoo, DeShedding Solution and Up to 20 Minutes Brushing with FURminator tool.
            </td>
            <td
              className="d-flex col-1 text-center justify-content-center align-self-center serviceCategoryPrice"
            >
              <p
                className="mr-2"
              >
                $
              </p>
              <p>
                81.00
              </p>
            </td>
          </div>
        </tr>
      </tbody>
      <hr
        className=" col-12 "
      />
    </table>
  </div>
</div>
`;
